src/main/java/de/fraunhofer/aisec/cpg/graph/type/TypeParser.java

```diff
@@ -263,6 +270,20 @@ private static boolean isUnknownType(String typeName) {
    */
   @NonNull
   private static String fixGenerics(@NonNull String type) {
+    if (type.contains("<") && type.contains(">")) {
+      String generics = type.substring(type.indexOf('<') + 1, type.lastIndexOf('>'));
+      for (String elaborated : elaboratedTypes) {
+        if (getLanguage() == TypeManager.Language.CXX && generics.contains(elaborated + " ")) {
+          generics = generics.replace(elaborated + " ", "");
```

Comment:
Not quite sure about this solution. Wouldn't the following fail?
`Array<myclass >`

Reply:
Can't we just pipe between `<>` through  `TypeParser.createFrom` BEFORE splitting the outer type? Then you would not need to remove white-spaces at all?